### DISCUSSION TITLE
Case sensitive URLs???

### DISCUSSION THREAD

**Discussion by jobadder-daviddancy** [Azure/data-api-builder#1332] [General]
Hi all!

This library looks amazing. I'm more than a little puzzled by the design decision to make REST URLs case-sensitive. Can anyone say why this decision was made this way please? It's the exact opposite of what I would have expected. Thanks!

**Comment by aaronburtle** [comment]
Hi @jobadder-daviddancy and thanks for the question!

The reason we decided to go with case-sensitivity is two-fold. Firstly, the IETF standard defines case-sensitivity for all portions of a URI aside from the `scheme` and `host`
https://datatracker.ietf.org/doc/html/[draft-ietf-httpbis-p1-messaging-25](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-p1-messaging-25#page-19)#page-19

![image](https://user-images.githubusercontent.com/93220300/225781437-016fd351-87c8-4173-8154-3a53c3024e25.png)

This stackoverflow has a great discussion about the topic you can read as well.
https://webmasters.stackexchange.com/questions/90339/why-are-urls-case-sensitive


And then we also have decided to keep case-sensitivity to better align with `GraphQL`, which uses case-sensitivity in its specifications. You can read more about `GraphQL` specs here:
https://github.com/graphql/graphql-spec/blob/34941384f26fa627b7aa1df7944bb24b0e7fb018/spec/Section%202%20--%20Language.md

Cheers,
Aaron
